### ISSUE TITLE
FCOS stable fails on -firstboot: "cannot change memory protections"

### ISSUE THREAD

**Issue by alexanderbystrom** [okd-project/okd#1414]
**Describe the bug**
Installation fails on openstack waiting for bootstrapping to complete. Tried several times and it always fails on the same step.

Latest working installer that I have tested is: 4.9.0-0.okd-2022-02-12-140851

**Version**
4.11.0-0.okd-2022-11-05-030711
IPI
OpenStack

**How reproducible**
100%

**Log bundle**
[log-bundle-20221118114318.tar.gz](https://github.com/okd-project/okd/files/10040593/log-bundle-20221118114318.tar.gz)

**Comment by vrutkovs** [comment]
```
Nov 18 10:13:43 okd-mcwr4-master-2.novalocal systemd[1]: Starting machine-config-daemon-pull.service - Machine Config Daemon Pull...
...
Nov 18 10:13:55 okd-mcwr4-master-2.novalocal peaceful_sutherland[1927]: /usr/bin/coreutils: error while loading shared libraries: librt.so.1: cannot change memory protections
```

In journal I see
```
Nov 18 10:13:57 okd-mcwr4-master-0.novalocal kernel: audit: type=1400 audit(1668766437.341:568): avc:  denied  { read } for  pid=1977 comm="cp" path="/usr/lib64/librt-2.28.so" dev="vda4" ino=9437580 scontext=system_u:system_r:container_t:s0:c87,c816 
```
and a lot of AVC warnings. Assuming that FCOS image is not modified and genuine - could it be storage permissions?

**Comment by alexanderbystrom** [comment]
The installer downloads the image so it's not modified (https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/36.20220716.3.1/x86_64/fedora-coreos-36.20220716.3.1-openstack.x86_64.qcow2.xz?sha256=f9c4309ee7a86161927fe16003fbb8cd8fe66343a186c8cbe7220f5963b1ee12)

What could the storage permission be?

**Comment by vrutkovs** [comment]
I assumed that underlying storage is misconfigured - but its a block storage and also bootstrap passes, so probably not related.

Checking if I can reproduce this on our CI

**Comment by Zanooda** [comment]
Having the same issue with 4.11.0-0.okd-2022-11-19-050030 on my OpenStack cluster.

**Comment by vrutkovs** [comment]
Also occurs on vSphere - https://github.com/okd-project/okd/discussions/1422

**Comment by GeoEducator** [comment]
What instructions where you using?

**Comment by alexanderbystrom** [comment]
> What instructions where you using?

Instructions?

**Comment by vrutkovs** [comment]
Seeing this popping up randomly in CI too, workaround: `rm -rf /var/lib/containers/storage`. The proper fix would be updating a base FCOS ISO

**Comment by mschauf** [comment]
> Seeing this popping up randomly in CI too, workaround: `rm -rf /var/lib/containers/storage`. The proper fix would be updating a base FCOS ISO

Connected to the bootstrap and tried to delete the directory.

```
[root@api ~]# rm -rf /var/lib/containers/storage
rm: cannot remove '/var/lib/containers/storage/overlay/76f0d448af6462deb8c0e3c4a0861f27f15f1daad2ebcc624b32f304b4dac9e8/merged': Device or resource busy
rm: cannot remove '/var/lib/containers/storage/overlay/dfb1b9fd494b322bd4f4ebf295b747a23e30c6276d92dba3507d671b8d20ce5c/merged': Device or resource busy
rm: cannot remove '/var/lib/containers/storage/overlay/84ebfb3fbd87a5f817229b1b138bb8aaeb21018ba23e590ebc5f3d431d15ecd6/merged': Device or resource busy
```


Have to stop the containers in the first place, like

```
[root@api ~]# crictl ps
CONTAINER           IMAGE                                                                                                   CREATED             STATE               NAME                             ATTEMPT             POD ID              POD
dacc49ffc642c       quay.io/openshift/okd-content@sha256:de6d75eb4a4ebb5c5953c0692e21b5e891857f45b3a6429ed58b63c41c4e1897   13 minutes ago      Running             cluster-policy-controller        0                   6957cd2d45577       bootstrap-kube-controller-manager-localhost.localdomain
89ce412c016dd       73cd98fa55ad9c78257773d72583f94b23c744a4632ba93a52cf2d0cdc8bb9cc                                        13 minutes ago      Running             kube-apiserver-insecure-readyz   0                   a8fbdb7e5e714       bootstrap-kube-apiserver-localhost.localdomain
8e1cc3a8af1b0       4dec8a86150d8d01ce4de4dbbf484bbc7ce77db60dc11a3ee2a927fe27bae2cc                                        13 minutes ago      Running             kube-apiserver                   0                   a8fbdb7e5e714       bootstrap-kube-apiserver-localhost.localdomain
6220177b0724f       quay.io/openshift/okd-content@sha256:b25fd3e811f854891ddfa8fe9a1bd1aa877b5a94e8e9f9eb6329f39ff96d31d0   13 minutes ago      Running             kube-scheduler                   0                   47a095f37ad58       bootstrap-kube-scheduler-localhost.localdomain
41ea89dcc816e       quay.io/openshift/okd-content@sha256:3f0ef4c3b44edb6fe61a1fdc4bdcecfcefbe71a7dedb5c1da7d778d0b19dcc1c   13 minutes ago      Running             coredns                          0                   318513a75e2a4       coredns-localhost.localdomain
6d74524f3dd43       quay.io/openshift/okd@sha256:924958ed5ba93ea9d72e4b5a319d0d4df220edbb891816c106215cbf318a0a40           13 minutes ago      Running             cluster-version-operator         0                   f9556dc762c03       bootstrap-cluster-version-operator-localhost.localdomain
4a62e2026cc61       bec2b7dffdf719874e01a1a7e750b6e908e6ceeb87bfb2744fa1ac4cfa78d999                                        13 minutes ago      Running             cloud-credential-operator        0                   9be2f6588b0fc       cloud-credential-operator-localhost.localdomain
33bd41a0c7c1f       quay.io/openshift/okd-content@sha256:7c7df602ae311bd172550bfba3b944ae157c145dd67e3ec02d081946c5d13077   13 minutes ago      Running             keepalived-monitor               0                   ddf75336a8d6d       keepalived-localhost.localdomain
f7654b5fe93e8       quay.io/openshift/okd-content@sha256:37b53b8b2591b9400a932561ff927271684e91ffc1b2a9842e01ec61f165dbfa   13 minutes ago      Running             keepalived                       0                   ddf75336a8d6d       keepalived-localhost.localdomain
b52054ee034a9       cdcd8d14a2606c85b34032ef315ddbdd446eee5788db90bcd7e77296a9cf2333                                        13 minutes ago      Running             machine-config-server            0                   a4a0a1e739287       bootstrap-machine-config-operator-localhost.localdomain
0286de415baaf       c092b31ec21b4fae7bcee5abb1fba1ba940a0799e4567f954297e34d32aa79cd                                        13 minutes ago      Running             etcd                             0                   eb5d0ffe7a1f6       etcd-bootstrap-member-localhost.localdomain
69812505c83a8       c092b31ec21b4fae7bcee5abb1fba1ba940a0799e4567f954297e34d32aa79cd                                        13 minutes ago      Running             etcdctl                          0                   eb5d0ffe7a1f6       etcd-bootstrap-member-localhost.localdomain
```

and then 
```
[root@api ~]# crictl stop dacc49ffc642c
E1219 11:29:34.712097    6632 remote_runtime.go:505] "StopContainer from runtime service failed" err="rpc error: code = DeadlineExceeded desc = context deadline exceeded" containerID="dacc49ffc642c"
FATA[0002] stopping the container "dacc49ffc642c": rpc error: code = DeadlineExceeded desc = context deadline exceeded
```

**Comment by Zanooda** [comment]
I was able to get the installation to complete using 4.11.0-0.okd-2022-12-02-145640 after setting

`platform:
  openstack:
    clusterOSImage: https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/36.20221030.3.0/x86_64/fedora-coreos-36.20221030.3.0-openstack.x86_64.qcow2.xz`

in my install-config.yaml - the issue seems indeed to be fixed in FCOS builds newer than the ones used by default by the installer.

**Comment by vrutkovs** [comment]
Thanks, we'll soon bump installer to use that image (but first [bump to F37 in 4.12+](https://issues.redhat.com/browse/OCPBUGS-1695) needs to land)

**Comment by alexanderbystrom** [comment]
> I was able to get the installation to complete using 4.11.0-0.okd-2022-12-02-145640 after setting
> 
> ```
> platform:
>   openstack:
>     clusterOSImage: https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/36.20221030.3.0/x86_64/fedora-coreos-36.20221030.3.0-openstack.x86_64.qcow2.xz
> ```
> 
> in my install-config.yaml - the issue seems indeed to be fixed in FCOS builds newer than the ones used by default by the installer.

I got an issue with the bootstrap node being unable to lookup api-int.clustername when testing with that image.

**Comment by mschauf** [comment]
> > I was able to get the installation to complete using 4.11.0-0.okd-2022-12-02-145640 after setting
> > ```
> > platform:
> >   openstack:
> >     clusterOSImage: https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/36.20221030.3.0/x86_64/fedora-coreos-36.20221030.3.0-openstack.x86_64.qcow2.xz
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > in my install-config.yaml - the issue seems indeed to be fixed in FCOS builds newer than the ones used by default by the installer.
> 
> I got an issue with the bootstrap node being unable to lookup api-int.clustername when testing with that image.

Same here - Cluster does not go up with

```
platform:
  vsphere:
    clusterOSImage: https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/latest/rhcos-4.11.9-x86_64-vmware.x86_64.ova
```

**Comment by Zanooda** [comment]
> I got an issue with the bootstrap node being unable to lookup api-int.clustername when testing with that image.

My workaround was to execute
`sudo systemctl restart systemd-resolved && sudo systemctl stop systemd-resolved`
on the bootstrap node.

**Comment by vrutkovs** [comment]
Resolved in latest 4.12 stable
